Clean up DNS example

talex5 opened the pull request:
- Split the code into smaller functions.
- Use logging API instead of passing a console around.
- Add README.
- Replace network configuration with generic_stackv4 (allows running
  with the native stack too).
- Move hard-coded settings to top of file.
- Comment out confusing/unlabelled hexdump.
- Don't send client test from source port that the server is listening
  on.
- Listen on port 53 by default. If a non-default port is used,
  `nslookup` must be run in interactive mode and in this mode it gets
  `SERVFAIL` for some reason.

yomimono commented:
It seems the Travis failures here are not a false positive (!):

```
File "unikernel.ml", line 71, characters 54-67:
Error: Unbound value pp_print_list
Did you mean pp_print_int?
Command exited with code 2.
```

samoht commented:
Can you rebase it on top of #126?

talex5 commented:
@samoht : rebased
@yomimono : looks like this was added in OCaml 4.02. I've updated the travis script to build against this - I think we all agreed to drop 4.01, and now is probably the time to do it. However, I could easily add a compatibility function if we want to support 4.01 too.

samoht commented:
Hum sorry you need to rebase again :-/

talex5 commented:
Merging before any more conflicts turn up ;-)
